### PR TITLE
log error message when the reach provider fails

### PR DESCRIPTION
**Pull request by d2fong**
- clean up require statements

- not sure what to return from fillDoc if there is an error, should I just return an empty doc?

### CODE REVIEW

**Comment by maxkfranz** [comment]
I don't think it should return an empty doc.  We could return an error json, but I think it would be best to just send an http error for now: https://expressjs.com/en/4x/api.html#res.sendStatus

**Comment by maxkfranz** [comment]
Probably 500?

**Comment by sacdallago** [comment]
Isn't this error generated IF there is an unexpected response from REACH? Aka: the response is generated, it's just not what we expect? And if so: isn't it impossible for the user to know what she/he has to change in order to _make it work_? If that is the case, not allowing the user to just _move on_ even if on an empty document might be frustrating to the user.

**Comment by d2fong** [comment]
Ok I send a 500 error when there there is a problem with creating a doc from reach.

**Comment by sacdallago** [comment]
I think this is a good start, but maybe we want to merge this in with all the other cases listed here: #207

File: src/server/routes/api/document/index.js [hunk 1]
```diff
@@ -73,6 +79,10 @@ http.post('/', function( req, res ){
     .then( runLayout )
     .then( getDocJson )
     .then( json => res.json( json ) )
+    .catch( e => {
+      logger.error('Could not fill doc from text: ', `text: ${text}`, e);
+      res.sendStatus(500);
```
Comment (maxkfranz) [inline src/server/routes/api/document/index.js]: Need to rethrow also

**Comment by d2fong** [comment]
Ok the build CI check has passed.  Weird that it failed before, but succeeded after I restarted the build.

**Comment by maxkfranz** [comment]
Travis doesn't seem to work well with RethinkDB.  Sometimes it just crashes.

**Comment by d2fong** [comment]
So can I merge this?

File: src/server/routes/api/document/index.js [hunk 2]
```diff
@@ -73,6 +79,16 @@ http.post('/', function( req, res ){
     .then( runLayout )
     .then( getDocJson )
     .then( json => res.json( json ) )
+    .catch( e => {
+
+      if( e instanceof Promise.TimeoutError ){
+        res.sendStatus(504);
+      } else {
+        logger.error('Could not fill doc from text: ', `text: ${text}`, e);
```
Comment (maxkfranz) [inline src/server/routes/api/document/index.js]: Did you text the output of this?  I'm not sure that Winston logs objects with a prototype like this properly.  Maybe it's best to use a single template string.  You could use `${e.toString()}` to guarantee you get a printable representation.

**Comment by maxkfranz** [comment]
Looks fine but I'm not sure about line 87.  I would have merged yesterday but didn't have time to investigate 87

File: src/server/routes/api/document/index.js (hunk 2 above)
Comment (maxkfranz) [reply src/server/routes/api/document/index.js]: Maybe better to have the error as a separate logger.error() call

**Comment by maxkfranz** [comment]
I've turned off Travis for older versions of Node.  It seems like 6 hangs sometimes on npm install.  This may help with some of the Travis noise.

We use Node 8 now anyway.

File: src/server/routes/api/document/index.js (hunk 2 above)
Comment (sacdallago) [reply src/server/routes/api/document/index.js]: I agree with the second call to error with just the response, first point needs testing, javascript is all but an exact science.

**Pull request merged**